src/cc/lib/graph/metadata.cc

```diff
@@ -73,6 +73,12 @@ Metadata::Metadata(std::filesystem::path path, std::string config_path, bool ski
         m_node_feature_count = 0;
         m_edge_feature_count = 0;
     }
+    // Skip watermark loading if requested. This is useful when the watermark is not needed for the training/inference
+    // job.
+    if (skip_watermark_loading == true)
```

Comment:
Watermark is already -1, we should skip trying to read it from the meta instead of reading it from meta then setting it to -1 again.

Reply:
Thanks, I updated the code to only load the meta if skip feature/temporal loading is not set to true.